ligo/math.mligo

```diff
@@ -95,16 +120,21 @@ let rec half_bps_pow_rec ((tick, acc, ladder) : nat * fixed_point * (fixed_point
         | [] -> (failwith price_out_of_bounds_err : fixed_point)
         | h :: t -> half_bps_pow_rec (half, (if rem = 0n then acc else fixed_point_mul h acc), t)
 
+(*
+  For a tick index `i`, calculate the corresponding `sqrt_price`:
+    sqrt(e^bps)^i * 2^80
+  using the exponentiation by squaring method, where:
+    bps = 0.0001
+ *)
 let half_bps_pow (tick : int) : x80n =
-    let product = half_bps_pow_rec (abs tick, {v=0n;offset=0}, (if tick > 0  then positive_ladder  else negative_ladder)) in
+    let product = half_bps_pow_rec (abs tick, {v=1n;offset=0}, (if tick > 0  then positive_ladder  else negative_ladder)) in
     let doffset = -80 - product.offset in
     if doffset > 0 then
-        {x80 = Bitwise.shift_right product.v (abs doffset)}
+        {x80 = stepped_shift_right (product.v, abs doffset)}
     else
         (* This branch should almost never happen, in general the price we get is not a round number. *)
         {x80 = Bitwise.shift_left product.v (abs doffset)}
```

Comment:
Despite being rare, isn't this also possibly going shift too much?

Also, if not, I suppose that `stepped_shift_left` could be removed, as it's not used anywhere

Reply:
Oh right, I forgot to use `stepped_shift_left` here! Great catch.